rest/view_api_test.go

```diff
@@ -221,47 +220,63 @@ func TestUserViewQuery(t *testing.T) {
 	a.Save(quinn)
 
 	// Have the user query the view:
-	request, _ := http.NewRequest("GET", "/db/_design/foo/_view/bar?include_docs=true", nil)
-	request.SetBasicAuth("quinn", "123456")
-	response = rt.Send(request)
-	assertStatus(t, response, 200)
-	var result sgbucket.ViewResult
-	json.Unmarshal(response.Body.Bytes(), &result)
+
+	result, err := rt.WaitForNUserViewResults(1, "/db/_design/foo/_view/bar?include_docs=true", quinn, password)
+	assertNoError(t, err, "Unexpected error")
 	assert.Equals(t, len(result.Rows), 1)
 	assert.Equals(t, result.TotalRows, 1)
 	row := result.Rows[0]
 	assert.Equals(t, row.Key, float64(7))
 	assert.Equals(t, row.Value, "seven")
-	assert.DeepEquals(t, *row.Doc, map[string]interface{}{"key": 7.0, "value": "seven", "channel": "Q"})
+
+	if base.UnitTestUrlIsWalrus() {
+		// include_docs=true only works with walrus as documented here:
+		// https://forums.couchbase.com/t/do-the-viewquery-options-omit-include-docs-on-purpose/12399
+		assert.DeepEquals(t, *row.Doc, map[string]interface{}{"key": 7.0, "value": "seven", "channel": "Q"})
+	}
 
 	// Admin should see both rows:
-	response = rt.SendAdminRequest("GET", "/db/_design/foo/_view/bar", ``)
-	assertStatus(t, response, 200)
-	json.Unmarshal(response.Body.Bytes(), &result)
+
+	result, err = rt.WaitForNAdminViewResults(2, "/db/_design/foo/_view/bar")
+	assertNoError(t, err, "Unexpected error")
 	assert.Equals(t, len(result.Rows), 2)
 	row = result.Rows[0]
 	assert.Equals(t, row.Key, float64(7))
 	assert.Equals(t, row.Value, "seven")
-	assert.DeepEquals(t, *row.Doc, map[string]interface{}{"key": 7.0, "value": "seven", "channel": "Q"})
+	if base.UnitTestUrlIsWalrus() {
+		// include_docs=true only works with walrus as documented here:
+		// https://forums.couchbase.com/t/do-the-viewquery-options-omit-include-docs-on-purpose/12399
+		assert.DeepEquals(t, *row.Doc, map[string]interface{}{"key": 7.0, "value": "seven", "channel": "Q"})
+	}
 	row = result.Rows[1]
 	assert.Equals(t, row.Key, float64(10))
 	assert.Equals(t, row.Value, "ten")
-	assert.DeepEquals(t, *row.Doc, map[string]interface{}{"key": 10.0, "value": "ten", "channel": "W"})
+
+	if base.UnitTestUrlIsWalrus() {
+		// include_docs=true only works with walrus as documented here:
+		// https://forums.couchbase.com/t/do-the-viewquery-options-omit-include-docs-on-purpose/12399
+		assert.DeepEquals(t, *row.Doc, map[string]interface{}{"key": 10.0, "value": "ten", "channel": "W"})
+	}
 
 	// Make sure users are not allowed to query internal views:
-	request, _ = http.NewRequest("GET", "/db/_design/sync_gateway/_view/access", nil)
-	request.SetBasicAuth("quinn", "123456")
+	request, _ := http.NewRequest("GET", "/db/_design/sync_gateway/_view/access", nil)
+	request.SetBasicAuth(quinn.Name(), password)
 	response = rt.Send(request)
 	assertStatus(t, response, 403)
 }
 
 // This includes a fix for #857
 func TestAdminReduceViewQuery(t *testing.T) {
 
+	if !base.UnitTestUrlIsWalrus() {
```

Comment:
Sanity check whether this works now that the wait has been added.  Otherwise I don't know why reduce shouldn't work against Couchbase Server (maybe get details on what exactly the 500 error is)

Reply:
This is still failing.  Documented in  https://github.com/couchbase/sync_gateway/issues/2977#issuecomment-337686467

Reply:
Looks like we still don't have the details on what error message is being returned w/ the 500 error on #2977.  Would be nice to understand why exactly it's failing, to understand whether it's a more serious issue.

Reply:
I'll look at that as part of #2977 and put that as a high priority given that it might be a serious issue.